src/Init/Data/String/Basic.lean

```diff
@@ -455,6 +455,10 @@ instance : Inhabited String := ⟨""⟩
 
 instance : Append String := ⟨String.append⟩
 
+@[simp]
+theorem String.empty_append (s : String) :
```

Comment:
You're already in the `String` namespace, so this will create `String.String.empty_append`.